orchestration/dagster_orchestration/hca_orchestration/tests/contrib/test_argo_workflows.py

```diff
@@ -0,0 +1,219 @@
+            self.assertEqual(list_results, archived_workflows + archived_workflows_2)
+            self.assertEqual(mock_list_archived_workflows.call_count, 2)
+
+            mock_list_archived_workflows.assert_has_calls([call(), call(list_options_continue="next_page")])
+
+    def test__pull_paginated_results_api_function_multiple_continues_results_multiple_calls(self):
+        client = ArgoArchivedWorkflowsClient("https://zombo.com", "tokentokentoken")
+        archived_workflows = [
+            mock_argo_workflow('import-hca-total-defg', 'abc123uid', 'Succeeded'),
+            mock_argo_workflow('import-hca-total-abcd', 'abc234uid', 'Supsneeded', params={
+                'data-repo-name': 'datarepo_dataset1'
+            }),
+            mock_argo_workflow('import-hca-total-cdef', 'abc345uid', 'Succeeded', params={
+                'data-repo-name': 'datarepo_dataset2'
+            }),
+        ]
+
+        test_mock = MagicMock()
+        test_mock.items = archived_workflows
+        test_mock.metadata._continue = "page2"
+
+        archived_workflows_2 = [
+            mock_argo_workflow('import-hca-total-page2', 'page2uid', 'Supsneeded', params={
+                'data-repo-name': 'datarepo_dataset1'
+            }),
+        ]
+
+        test_mock2 = MagicMock()
+        test_mock2.items = archived_workflows_2
+        test_mock2.metadata._continue = "page3"
+
+        archived_workflows_3 = [
+            mock_argo_workflow('import-hca-1', '12345', 'Succeeded', params={
+                'data-name': 'dataset',
+                'test': 'data'
+            }),
+            mock_argo_workflow('hca-data', 'u123id', 'Succeeded'),
+        ]
+
+        test_mock3 = MagicMock()
+        test_mock3.items = archived_workflows_3
+        test_mock3.metadata._continue = "page4"
+
+        archived_workflows_4 = [
+            mock_argo_workflow('104', 'uid', 'Succeeded', params={
+                'dataset': 'data'
+            }),
+        ]
+
+        test_mock4 = MagicMock()
+        test_mock4.items = archived_workflows_4
+        test_mock4.metadata._continue = None
+
+        def get_page(list_options_continue=None):
+            if list_options_continue is None:
+                return test_mock
+
+            if list_options_continue == "page2":
+                return test_mock2
+
+            if list_options_continue == "page3":
+                return test_mock3
+
+            if list_options_continue == "page4":
+                return test_mock4
+
+        with patch('argo.workflows.client.ArchivedWorkflowServiceApi.list_archived_workflows',
+                   side_effect=get_page) as mock_list_archived_workflows:
+            list_results = list(client._pull_paginated_results(mock_list_archived_workflows))
+            self.assertEqual(list_results, archived_workflows + archived_workflows_2 + archived_workflows_3 +
+                             archived_workflows_4)
```

Comment:
nit: when spreading function arguments across multiple lines, avoid putting two arguments on the same line (i always add a newline after each comma when doing this).